pkg/statedb2/metrics.go

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package statedb2
+
+import (
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+)
+
+type Metrics struct {
+	// How long a read transaction was held.
+	WriteTxnDuration metric.Vec[metric.Observer]
+	// How long it took to acquire a write transaction for all tables.
+	WriteTxnAcquisition metric.Vec[metric.Observer]
+	// How long writers were blocked while waiting to acquire a write transaction for a specific table.
+	TableContention metric.Vec[metric.Gauge]
+	// The amount of objects in a given table.
+	TableObjectCount metric.Vec[metric.Gauge]
+	// The current revision of a given table.
+	TableRevision metric.Vec[metric.Gauge]
+	// The amount of delete trackers for a given table.
+	TableDeleteTrackerCount metric.Vec[metric.Gauge]
+	// The amount of objects in the graveyard for a given table.
+	TableGraveyardObjects metric.Vec[metric.Gauge]
+	// The lowest revision of a given table that has been processed by the graveyard garbage collector.
+	TableGraveyardLowWatermark metric.Vec[metric.Gauge]
+	// The time it took to clean the graveyard for a given table.
+	TableGraveyardCleaningDuration metric.Vec[metric.Observer]
+}
+
+func NewMetrics() Metrics {
```

Comment:
Yea, I see what you are saying, its a concern. A lot of these metrics are to track things that we as developers are worried about. The idea being that if something goes wrong and we get a sysdump that we can hopefully see if certain things are going wrong in statedb. 

But metrics might not be the correct tool for the job, or at least not in this way. Perhaps we should look into a setting where metrics are part of the CLI output and thus the sysdump and the prometheus server.

Reply:
I think these metrics may be potentially useful to some users.  It may also help us reveal scaling issues in the future.

This is kind of part of a broader problem we have that I've been thinking about.  We find ourselves in a position where we walk a line between offering metrics that are high cardinality and providing ones that can be enabled by default, or we make compromises (i.e. see map_pressure metric).

Not sure what the answer is here, I think in this case we may just want to think about what we enable by default.